### PR TITLE
60 seed data

### PR DESCRIPTION
**Pull request by ski-climb**
@tmikeschu @njgheorghita @Sh1pley @riverswb Don't merge this one yet, but definitely take a look.

Worth noting: this branch will require the branch that @riverswb has (removing the reservations foreign key on the nights model) be merged in before this one.

Soooo, after I pulled the trigger and merged the branch for making reservations, I realized that, although the tests passed, none of it worked in my browser.  Many sleuthings later... it seems that the date-picker that we're using provides dates in the mm/dd/yyyy format, but we set up some of our tests before we had the date-picker in place, so we were dealing with dates in the dd-mm-yyyy format.  Very understandable on both counts, but now we have to make it work.

Bottom line is the `Date.parse(some-date)` doesn't play well with mm/dd/yyyy format that we get from the date-picker.

The commit `792de3c` makes the change in our code to the mm/dd/yyyy format for passing around dates (so that the tests will more accurately represent what's happening in the browser).  This broke many of the specs, so those were rewritten too.

It came down to a choice of either letting the `Night` model accept only a very restricted date format, or assume that the `Night` model would always be provided with check_in and check_out dates which were `Date` objects or played well with `Date.parse(some_date)`.  I chose the latter, but please let me know if you think the trade-off makes more sense the other way.  I'm open minded.

### CODE REVIEW

File: Gemfile
Comment (tmikeschu) [inline Gemfile:29]: @ski-climb shiny buttons! What does it do?

File: app/models/couch.rb
Comment (tmikeschu) [inline app/models/couch.rb:22]: Totally cool with this solution 👍

File: app/models/night.rb
Comment (tmikeschu) [inline app/models/night.rb:9]: @ski-climb what happens without this coercion?

File: db/seeds.rb
Comment (tmikeschu) [inline db/seeds.rb:16]: Clever!

File: db/seeds.rb
Comment (tmikeschu) [inline db/seeds.rb:36]: 👍 thinking ahead

File: db/seeds.rb
Comment (tmikeschu) [inline db/seeds.rb:29]: Can't wait to see this progress bar in action!

File: db/seeds.rb
Comment (tmikeschu) [inline db/seeds.rb:6]: 👌

File: db/seeds.rb
Comment (tmikeschu) [inline db/seeds.rb:49]: @ski-climb you had too much fun 😄

File: db/seeds.rb
Comment (tmikeschu) [inline db/seeds.rb:104]: @ski-climb can we do an in person walk through on this tomorrow? Looks solid, just want to get a more in depth explanation.

File: spec/features/reservations/traveler_can_make_a_reservation_spec.rb
Comment (tmikeschu) [inline spec/features/reservations/traveler_can_make_a_reservation_spec.rb:33]: 👍

File: spec/models/reservation_maker_spec.rb
Comment (tmikeschu) [inline spec/models/reservation_maker_spec.rb:11]: Thanks for improving the readability as well!

**Review by tmikeschu — COMMENTED** [review]
Super thorough fix on the date problem, and nice work with the seed!

Ready to merge once @riverswb 's PR is in.

File: app/models/couch.rb
Comment (njgheorghita) [reply app/models/couch.rb:22]: 👍

File: db/seeds.rb
Comment (njgheorghita) [inline db/seeds.rb:18]: Lolz

File: db/seeds.rb
Comment (njgheorghita) [reply db/seeds.rb:104]: ditto, would also find that helpful

**Comment by njgheorghita** [comment]
Killer @ski-climb, thanks for taking care of this, definitely gonna be a massive help with dev and production. Not sure where we are on the merge status, still waiting for the night model pr?? either way it's good to merge asap

File: Gemfile
Comment (ski-climb) [reply Gemfile:29]: When you run `rails db:seed` it will show a progress bar across your terminal as it creates new objects in the database.  The whole process takes... maybe 20-30 seconds?

File: app/models/night.rb
Comment (ski-climb) [reply app/models/night.rb:9]: This might be a bit unnecessary.  Basically it allows us to pass strings (e.g. "2017-02-22") OR Date objects (e.g. `Date.current`) to the scope.  I was thinking about programming for least surprise, and it just felt right that this scope should be able to handle Date objects.

File: db/seeds.rb
Comment (ski-climb) [reply db/seeds.rb:104]: @tmikeschu @njgheorghita happy to!

File: spec/features/reservations/traveler_can_make_a_reservation_spec.rb
Comment (tmikeschu) [inline spec/features/reservations/traveler_can_make_a_reservation_spec.rb:13]: @ski-climb I think this can be deleted because of your rails helper setup.

**Review by tmikeschu — APPROVED** [review]
Nice work, @ski-climb ! See comment about needing to include your model helpers module. 


Thumbs up for the next reviewer to merge.

**Review by riverswb — APPROVED** [review]
Great work! Merging

File: spec/features/reservations/traveler_can_make_a_reservation_spec.rb
Comment (ski-climb) [reply spec/features/reservations/traveler_can_make_a_reservation_spec.rb:13]: I thought so too... but then I was getting errors for not having the `.to_date_picker_format` method defined.  Perhaps I have it setup wrong.  Another set of eyes would be much appreciated!

**Pull request merged**
